math/mathcore/inc/Math/MinimizerOptions.h

```diff
@@ -28,23 +28,107 @@ class IOptions;
+    It contains also static methods for setting the default Minimizer option values
+    that will be used by default by all Minimizer instances.
+    To see the current default options do:
+
+         ROOT::Math::MinimizerOptions::PrintDefault();
+
 */
 class MinimizerOptions {
 
 public:
 
    // static methods for setting and retrieving the default options
 
+   /// Set the the default Minimizer type and corresponding algorithms 
+   /// Here is the list of the available minimizers and their corresponding algorithms. 
+   /// For some minimizers (e.g. Fumili) there are no specific algorithms available, then there is no need to specify it.
+   /// 
+   /// \anchor ROOTMinimizers
+   /// ### ROOT Minimizers
+   ///
+   /// - Minuit  Minimizer based on TMinuit, the legacy Minuit implementation. Here are the available algorithms:
+   ///     - Migrad     default algorithm based on the variable metric minimizer
+   ///     - Minimize   combination of Simplex and Migrad
+   ///     - Simplex    minimization algorithm not using the gradient information
+   ///     - Scan       brute function scan
+   /// - Minuit2  New C++ impelmentation of Minuit (the reccomended one)
+   ///     - Migrad  (default)
+   ///     - Minimize
+   ///     - Simplex
+   ///     - Fumili2    new impelmentation of Fumili integrated in Minuit2
+   /// - Fumili  Minimizer using an approximation for the Hessian based on first derivatives of the model function (see TFumili). Works only for chi-squared and likelihood functions.
+   /// - Linear  Linear minimizer (fitter) working only for linear functions (see TLinearFitter and TLinearMinimizer)
+   /// - GSLMultiMin  Minimizer from GSL based on the ROOT::Math::GSLMinimizer. Available algorithms are: 
+   ///    - BFGS2  (default)
+   ///    - BFGS
+   ///    - ConjugateFR
+   ///    - ConjugatePR
+   ///    - SteepestDescent
+   /// - GSLMultiFit Minimizer based on GSL for minimizing only non linear least-squared functions (using an approximation similar to Fumili). See ROOT::Math::GSLMultiFit.
+   /// - GSLSimAn Simulated annealing minimizer from GSL (see ROOT::Math::GSLSimAnMinimizer). It is a stochastic minimization algorithm using only function values and not the gradient.
+   /// - Genetic Genetic minimization algorithms (see TMVA::Genetic)
+   ///
    static void SetDefaultMinimizer(const char * type, const char * algo = 0);
+
+   /// Set the default level for computing the parameter errors 
+   /// For example for 1-sigma parameter errors
+   ///  - up = 1 for a chi-squared function
+   ///  - up = 0.5 for a negative log-likelihood function
+   ///
+   /// The value will be used also by Minos when computing the confidence interval
    static void SetDefaultErrorDef( double up);
+   
+   /// Set the Minimization tolerance. 
+   /// The Default value for Minuit and Minuit2 is 0.01
    static void SetDefaultTolerance(double tol);
+ 
+   /// Set the default Minimizer precision 
```

Comment:
```suggestion
   /// Set the default Minimizer precision.
```
(also in more functions in this file)